usr/sbin/pkcsslotd/mutex.c

```diff
@@ -27,35 +27,38 @@ int CreateXProcLock(void)
     struct group *grp;
     mode_t mode = (S_IRUSR | S_IRGRP);
 
-    if (xplfd == -1)
-        xplfd = open(OCK_API_LOCK_FILE, O_RDONLY);
-
     if (xplfd == -1) {
-        xplfd = open(OCK_API_LOCK_FILE, O_CREAT | O_RDONLY, mode);
-
-        if (xplfd != -1) {
-            if (fchmod(xplfd, mode) == -1) {
-                DbgLog(DL0, "%s:fchmod(%s):%s\n",
-                       __func__, OCK_API_LOCK_FILE, strerror(errno));
-                goto error;
-            }
+        /* Linux implementation of lockf doesn't care about the mode the file
+         * is opened in, but AIX does - to obtain exclusive locks, AIX needs
+         * the file to be opened in write mode. Minority wins.
+         * Also fix racy file creation by adding O_EXCL - this guarantees only
+         * one process succeeds in creating this lock file, which was likely
+         * the original intention.
+         */
+        xplfd = open(OCK_API_LOCK_FILE, O_CREAT | O_WRONLY | O_EXCL, mode);
```

Comment:
The daemon username is only changed on AIX, and is set to `slotd` since the first release. So that will not be a problem, but the other part makes sense. I'll revert these changes then.